source/server/server.cc

```diff
@@ -127,6 +127,8 @@ void InstanceImpl::drainListeners() {
 
 void InstanceImpl::failHealthcheck(bool fail) {
   // We keep liveness state in shared memory so the parent process sees the same state.
+  // TODO TODO TODO OH NO, CHILD->PARENT COMMUNICATION............ i guess this is fine, just make
```

Comment:
nit: Please rephrase "TODO(...): ..." with an actionable statement. On the liveness topic, my recommendation here is this is one of the stats that does not transfer from the parent. (I'm not sure what your plans are there exactly in how to indicate that.)

Reply:
Ha, whoops.

Is it really ok to just let the parent-child sharing of this state go away, though? The comment sounds like it's expected that the child is the one handling the healthcheck fail command, and the use of shared memory is the only way for the parent to get this update.

(I currently have a simple stat name->transfer logic enum map for exceptions to the "sensible default" transfer logic, with the map being hardcoded with the relevant exceptions. Good at least for getting the implementation up and running for review; maybe we'll want to switch to a more robust setup, like having the behavior specified in the stat definition macros.)